File: demo-android/app/src/androidTest/java/com/example/demo/e2e/support/mock/ApiRouteMockClient.kt
package com.example.demo.e2e.support.mock

import com.example.demo.e2e.support.mock.ApiRouteMock

class ApiRouteMockClient(
    private val wiremockHost: String = WireMockClient.resolveWiremockHost(),
) : AutoCloseable {
    private val wireMock = WireMockClient(wiremockHost)
    private var apiRouteMock: ApiRouteMock? = null

    fun api(): ApiRouteMock {
        if (apiRouteMock == null) {
            apiRouteMock = ApiRouteMock(wireMock)
        }
        return apiRouteMock!!
    }

    fun start() {
        reset()
        api().getTasks()
    }

    fun reset() {
        close()
    }

    override fun close() {
        wireMock.reset()
        apiRouteMock = null
    }
}

File: demo-ios/DemoUITests/Support/Mock/ApiRouteMockClient.swift
final class ApiRouteMockClient {
    private let wireMock = WireMockClient()
    private var apiMock: ApiRouteMock?

    func api() -> ApiRouteMock {
        if apiMock == nil {
            apiMock = ApiRouteMock(wireMock: wireMock)
        }
        return apiMock!
    }

    func start() throws {
        try reset()
    }

    func reset() throws {
        try wireMock.reset()
        apiMock = nil
    }
}
